### PR TITLE
Support net6.0 directly, dropping net5.0

### PR DESCRIPTION
**Pull request by blairconrad**
As mentioned in #1935, we intend to support certain legacy TFMs as well as the LTS .net releases.

### CODE REVIEW

**Comment by blairconrad** [comment]
Rationalized names of build profiles while I was here. Can split out if desired.

File: tests/FakeItEasy.Specs/IllegalFluentApiSpecs.cs [hunk 1]
```diff
@@ -147,15 +147,8 @@ public static void IllegalSyntax(string illegalOperation, string expectedError,
                 .x(() => diagnostics.Where(d => d.Severity == DiagnosticSeverity.Error).Should().ContainSingle()
                     .Which.GetMessage().Should().Contain(expectedError));
 
-#if MISMATCH_COMPILE_AND_RUNTIME_FRAMEWORKS
-            "And there are warnings due to a mistmach between compile-time and runtime frameworks"
-                .x(() => diagnostics.Where(d => d.Severity != DiagnosticSeverity.Error).Should().OnlyContain(d =>
-                    d.Severity == DiagnosticSeverity.Warning &&
-                    d.Id == "CS1701"));
-#else
```
Comment (thomaslevesque) [inline tests/FakeItEasy.Specs/IllegalFluentApiSpecs.cs:155]: I don't remember this. Was this an error that only occurred in .NET 6.0?

**Review by thomaslevesque — COMMENTED** [review]
Thanks @blairconrad!
Looks good to me, but I think we should also update how_to_build.md and README.md (which were also missing updates for the upgrade to net462)

File: tests/FakeItEasy.Specs/IllegalFluentApiSpecs.cs (hunk 1 above)
Comment (blairconrad) [reply tests/FakeItEasy.Specs/IllegalFluentApiSpecs.cs:155]: Yeah. Because we built with 5.0 and tested with 6.0.

**Comment by blairconrad** [comment]
> Thanks @blairconrad! Looks good to me, but I think we should also update how_to_build.md and README.md (which were also missing updates for the upgrade to net462)

Oh, of course. My search was weak!

**Comment by blairconrad** [comment]
> > Thanks @blairconrad! Looks good to me, but I think we should also update how_to_build.md and README.md (which were also missing updates for the upgrade to net462)
> 
> Oh, of course. My search was weak!

(Updating these is a pain!)

**Comment by blairconrad** [comment]
(I have a typo in the first 2 commit messages, so made their amendments squashes instead of fixups.)

File: how_to_build.md [hunk 2]
```diff
@@ -100,10 +100,10 @@ available:
 
 * `full`: the default profile, builds all supported target frameworks supported
   on the current platform
-* `net462`: builds only the .NET Framework 4.6.2 target framework
+* `net462`: builds and tests only the .NET Framework 4.6.2 target framework
```
Comment (thomaslevesque) [inline how_to_build.md:103]: Just thought of something. Maybe we should rename this profile to `netfx` or `netframework` or something similar. This way, if we change the .NET Framework version again, we won't have to rename it.

**Review by thomaslevesque — APPROVED** [review]
Thanks for the updates! Looks good to me now.
I just have one small suggestion; not sure if it's a good idea, let me know what you think!

File: how_to_build.md (hunk 2 above)
Comment (thomaslevesque) [reply how_to_build.md:103]: True. Let's leave it as is, then.

**Review by thomaslevesque — APPROVED** [review]
Thanks, @blairconrad!

**Comment by blairconrad** [comment]
Thank you, sir.

**Pull request merged**